toolbox/repo/validate_role_vars_used.py

```diff
@@ -0,0 +1,78 @@
+#! /usr/bin/env python
+
+import glob, os
+import yaml
+import subprocess
+import pathlib
+
+THIS_DIR = pathlib.Path(__file__).absolute().parent
+
+# go to top directory
+os.chdir(os.path.join(THIS_DIR, "..", ".."))
+
+ROLES_VARS_GLOB = "roles/*/vars/*/*"
+ROLES_DEFAULTS_GLOB = "roles/*/defaults/*/*"
+
+def validate_role_vars_used(filename, yaml_doc):
+    errors = 0
+
+    role_name = filename.split("/", 2)[1]
+    for key in yaml_doc.keys():
+        proc = subprocess.run(["grep", key, f"roles/{role_name}",
+                               "--recursive",
+                               ], capture_output=True)
+
+        if proc.returncode != 0:
+            # grep should always find 'key' inside 'filename'
+            # I couldn't find how to exclude one this file with grep options.
```

Comment:
Looks like this sentence got messed up:
`# I couldn't find how to exclude one this file with grep options.`
What did you mean?

Reply:
I meant: I couldn't find how to exclude one file with grep options. I wanted to do this:
```
KEY=entitlement_rhsm
ROLE=roles/entitlement_deploy
DEF_FILE=roles/entitlement_deploy/vars/main/resources.yml
grep $KEY $ROLE --exclude=DEF_FILE -r
```
but this doesn't work, this line still appears in the output:
```
roles/entitlement_deploy/vars/main/resources.yml:entitlement_rhsm: 'roles/entitlement_deploy/files/rhsm.conf'
```
(just in case you know how to do it with `grep` only)

Reply:
I see. Looks like grep `--exclude` is for file name patterns, and those patterns can't include directories. Useless.

Reply:
yes exactly, `--exclude=resources.yml` worked as expected, but that may exclude too many files!